vic/drivers/shared_image/include/vic_driver_shared_image.h

```diff
@@ -160,56 +160,55 @@ typedef struct {
  * @brief   file structures
  *****************************************************************************/
 typedef struct {
-    FILE *forcing[MAX_FORCE_FILES];   /**< forcing data files */
     FILE *globalparam;  /**< global parameters file */
     FILE *constants;    /**< model constants parameter file */
-    FILE *domain;       /**< domain file */
-    FILE *init_state;   /**< initial model state file */
-    FILE *paramfile;    /**< parameter file */
-    FILE *statefile;    /**< output model state file */
     FILE *logfile;      /**< log file */
 } filep_struct;
 
 /******************************************************************************
  * @brief   This structure stores input and output filenames.
  *****************************************************************************/
 typedef struct {
-    char forcing[MAX_FORCE_FILES][MAXSTRING];    /**< atmospheric forcing data file names */
+    nameid_struct forcing[MAX_FORCE_FILES];    /**< atmospheric forcing data file names */
     char f_path_pfx[MAX_FORCE_FILES][MAXSTRING]; /**< path and prefix for atmospheric forcing data file names */
     char global[MAXSTRING];        /**< global control file name */
-    char domain[MAXSTRING];        /**< domain file name */
+    nameid_struct domain;        /**< domain file name and nc_id*/
     char constants[MAXSTRING];     /**< model constants file name */
-    char params[MAXSTRING];        /**< model parameters file name */
-    char init_state[MAXSTRING];    /**< initial model state file name */
+    nameid_struct params;        /**< model parameters file name and nc_id */
+    nameid_struct init_state;    /**< initial model state file name and nc_id */
     char result_dir[MAXSTRING];    /**< directory where results will be written */
     char statefile[MAXSTRING];     /**< name of file in which to store model state */
     char log_path[MAXSTRING];      /**< Location to write log file to */
 } filenames_struct;
 
-void add_nveg_to_global_domain(char *nc_name, domain_struct *global_domain);
+void add_nveg_to_global_domain(nameid_struct nc_nameid,
+                               domain_struct *global_domain);
 void alloc_force(force_data_struct *force);
 void alloc_veg_hist(veg_hist_struct *veg_hist);
 double air_density(double t, double p);
 double average(double *ar, size_t n);
 void check_init_state_file(void);
-void compare_ncdomain_with_global_domain(char *ncfile);
+void compare_ncdomain_with_global_domain(nameid_struct nc_nameid);
 void free_force(force_data_struct *force);
 void free_veg_hist(veg_hist_struct *veg_hist);
 void get_domain_type(char *cmdstr);
-size_t get_global_domain(char *domain_nc_name, char *param_nc_name,
+size_t get_global_domain(nameid_struct domain_nc_nameid,
+                         nameid_struct param_nc_nameid,
                          domain_struct *global_domain);
```

Comment:
All of these should be `nameid_struct *`. Within the function you would then use a `->` rather than a `.`